ipn/ipnlocal/local.go

```diff
@@ -6455,6 +6456,37 @@ func (b *LocalBackend) FileTargets() ([]*apitype.FileTarget, error) {
 	return ret, nil
 }
 
+func (b *LocalBackend) taildropTargetStatus(p tailcfg.NodeView) ipnstate.TaildropTargetStatus {
+	if b.netMap == nil || b.state != ipn.Running {
+		return ipnstate.TaildropTargetIpnStateNotRunning
+	}
+	if b.netMap == nil {
+		return ipnstate.TaildropTargetNoNetmapAvailable
+	}
+	if !b.capFileSharing {
+		return ipnstate.TaildropTargetMissingCap
+	}
+
+	if !p.Valid() {
+		return ipnstate.TaildropTargetInvalidPeer
```

Comment:
nit: the check for Valid() here is really just looking to see if we have any info for the peer. If `p` were a pointer, it would just be nil. So, maybe something like `TaildropTargetNoPeerInfo` would be more accurate.